src/server/chat/commands.js

```diff
@@ -1,44 +1,17 @@
 const game = require("../game");
-const skins = require("../skins");
 
 const commands = function() {
-	const _colorRegEx = /#(?:[0-9a-fA-F]{3}){1,2}$/g;
-	const _createMarbleAttributesObject = function(messageContent) {
-		let messageSections = messageContent.split(" ");
-		let color, skinId;
-
-		// At most, check only the first three words. The first occurrence of an attribute will be used, the rest will be ignored.
-		for (let i = 1; i < Math.min(messageSections.length, 4); i++) {
-			if (!color) {
-				let match = messageSections[i].match(_colorRegEx);
-				color = (match === null ? undefined : match[0]);
-				if (typeof color !== "undefined") continue;
-			}
-
-			if (!skinId) {
-				// Get the skinId that corresponds to the provided message. Non-existing skins will set `skinId` to undefined.
-				skinId = skins.idList[messageSections[i]];
-				if (typeof skinId !== "undefined") continue;
-			}
-		}
-
-		return {
-			color,
-			skinId
-		};
-	};
-
 	return [
 		// Anyone
 		{
 			// Enter a marble
 			commandString: "marble",
 			permission: "ANYONE",
 			action: function(messageContent, id, username) {
-				game.addPlayerEntry(
+				game.addRaceEntry(
 					id,
 					username,
-					_createMarbleAttributesObject(messageContent)
+					messageContent.substr(7)
```

Comment:
Didn't catch `substr()` being deprecated, so that will be changed. I will also change it for `!level`, since this isn't the first time this function is used in this file.

`substring()` returns the string starting from the given index (and optionally, and end index). So here, it removes `!marble` from the message since the command isn't part of the attributes (to add to that, `addRaceEntry` isn't aware of commands and requests only a string of attributes). I'll change it to base it off of the command string directly, and add a comment to clarify the purpose.

(I'm not sure what kind of assumption was being made here in the end, but if you believed this code was flawed to the point where it would undoubtedly break functionality, *please* confirm this by running the code next time. There are cleaner and more to the point ways to comment on a magic number and/or a unclear function call)

Reply:
I'll also change the `substr()` calls for the rest of the codebase while I'm at it, since it's used more frequently than just here and well, it's deprecated.